### PR TITLE
Add marker based pagination for get users methods

### PR DESCRIPTION
**Pull request by sujaygarlanka**
### Issue Link :link:
- https://jira.inside-box.net/browse/SDK-1213

### Goals :soccer:
- Add marker based pagination for get users methods

### Implementation Details :construction:
- Added usemarker and marker parameters to multiple get users methods
- Updated BoxResourceIterable to be able to pass in marker to start pagination
- Updated BoxResourceIterable to expose the next marker string

### Testing Details :mag:
- Unit testing
- Manual testing

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @sujaygarlanka has signed the CLA. Thanks for the pull request!

File: src/main/java/com/box/sdk/BoxResourceIterable.java [hunk 1]
```diff
@@ -66,6 +71,26 @@ public BoxResourceIterable(BoxAPIConnection api, URL url, int limit) {
...
+        this.iterator = new IteratorImpl(null);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param api
+     *            the API connection to be used by the resource
+     * @param url
+     *            to end-point with paging support
```
[the remark this answers is not in the document]
Comment (sujaygarlanka) [reply src/main/java/com/box/sdk/BoxResourceIterable.java]: not sure. just copied this from above in the code.

File: src/main/java/com/box/sdk/BoxResourceIterable.java [hunk 2]
```diff
@@ -52,20 +52,45 @@
...
+    private final IteratorImpl iterator;
+
     /**
      * Constructor.
      *
      * @param api
      *            the API connection to be used by the resource
      * @param url
-     *            to end-point with paging support
+     *            end-point with paging support
```
Comment (PJSimon) [inline src/main/java/com/box/sdk/BoxResourceIterable.java]:
```suggestion
     *            endpoint with paging support
```

File: src/main/java/com/box/sdk/BoxResourceIterable.java [hunk 3]
```diff
@@ -52,20 +52,45 @@
...
+        this.iterator = new IteratorImpl(null);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param api
+     *            the API connection to be used by the resource
+     * @param url
+     *            end-point with paging support
```
Comment (PJSimon) [inline src/main/java/com/box/sdk/BoxResourceIterable.java]:
```suggestion
     *            endpoint with paging support
```

**Pull request merged**